src/components/MobileApp/DesignComponents/DatePicker.js

```diff
@@ -0,0 +1,152 @@
+  useEffect(() => {
+    const validDateFromProp = tryParseDate(value)
+    if (!isSameDay(validDateFromProp, validDate)) {
+      setValidDate(tryParseDate(value))
+    }
+  }, [value])
+
+  // Hide calendar on outside-click
+  useEffect(() => {
+    if (document) {
+      document.addEventListener('mousedown', handleClickOutside)
+      return () => {
+        document.removeEventListener('mousedown', handleClickOutside)
+      }
+    }
+  }, [])
+
+  // Validator
+  useEffect(() => {
+    let msg = '' // (Note: message is not displayed in UI)
+    if (value) {
+      const date = parseDate(value)
+      if (!isValid(date)) {
+        msg = 'unparseable'
+      } else if (minDate && isBefore(date, minDate) && !isSameDay(date, minDate)) {
+        msg = 'before minDate'
+      } else if (maxDate && isAfter(date, maxDate) && !isSameDay(date, maxDate)) {
+        msg = 'after maxDate'
+      }
+    }
+    inputRef.current && inputRef.current.setCustomValidity(msg)
+  }, [value])
+
+  return (
+    <div
+      className={cx('date-picker', { 'was-validated': wasValidated, 'date-picker--calendar-active': calendarActive })}
+      ref={containerRef}
+    >
+      <LabelInside>
+        <input
+          ref={inputRef}
+          className={cx('form-control calendar-indicator', className)}
+          value={value}
+          onFocus={handleFocus}
+          onBlur={handleBlur}
+          onChange={handleChange}
+          required={required}
+          autoComplete="off"
+          {...restProps}
+        />
+        {label}
+      </LabelInside>
+      <div className="leihs-calendar rounded-bottom">
+        <Calendar
+          key={validDate || 1} // force update shown date if changed via text input
+          displayMode="date"
+          className="m-0 w-100 rounded-bottom bg-light-shade"
+          locale={locale}
+          showDateDisplay={false}
+          date={validDate}
+          shownDate={validDate}
+          onChange={handleCalendarDateChange}
+          minDate={minDate}
+          maxDate={maxDate}
+          showMonthAndYearPickers={false}
+        />
+      </div>
+    </div>
+  )
+}
+
+DatePicker.propTypes = {
+  /** Locale for formatting and parsing of date string, e.g. `locale={de}` when `import { de } from 'date-fns/locale'` */
+  locale: PropTypes.object,
+  /** same as `HTMLInputElement` */
+  value: PropTypes.string,
+  /** same as `HTMLInputElement` */
+  onChange: PropTypes.func.isRequired,
+  /** same as `HTMLInputElement` */
+  required: PropTypes.bool,
+  /** the label (shown inside the input field) */
+  label: PropTypes.node,
+  /** same as `HTMLInputElement` */
+  className: PropTypes.string,
+  /** earliest date that can be selected */
+  minDate: PropTypes.instanceOf(Date),
+  /** latest date that can be selected */
+  maxDate: PropTypes.instanceOf(Date),
+  /** all other props will be applied to the input field (e.g. id, name, placeholder...) */
+  restProps: PropTypes.object
```

Comment:
that's wrong. `restProps` is not a prop name, just the name we internally use for the rest parameter. 

the line can just be removed, we can not (and dont need to) express this in terms of `propTypes` (there is no requirement for the list to be exhaustive). technically we could list all the DOM props we support but thats handled already be React (and quickly becomes complex when also allowing a prop to set the wrapper element, like `as="div"`)